python/TestHarness/RunParallel.py

```diff
@@ -156,7 +165,18 @@ def returnToTestHarness(self, job_index):
     did_pass = True
 
     output = 'Working Directory: ' + tester.specs['test_dir'] + '\nRunning command: ' + command + '\n'
-    output += self.readOutput(f)
+
+    if tester.specs.isValid('redirect_output'):
```

Comment:
Then you don't need the same condition in both `else` clauses. Just one `if` statement.